core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go

```diff
@@ -220,6 +221,41 @@ func (vdb *versionedDB) GetFullScanIterator(skipNamespace func(string) bool) (st
 	return newFullDBScanner(vdb.db, skipNamespace)
 }
 
+// ImportState implements method in VersionedDB interface. The function is expected to be used
+// for importing the state from a previously snapshotted state. The parameter itr provides access to
+// the snapshotted state.
+func (vdb *versionedDB) ImportState(itr statedb.FullScanIterator, dbValueFormat byte) error {
+	if dbValueFormat != fullScanIteratorValueFormat {
+		return errors.Errorf("value format [%x] not supported. Expected value format [%x]",
+			dbValueFormat, fullScanIteratorValueFormat)
+	}
+	dbBatch := vdb.db.NewUpdateBatch()
+	batchSize := 0
+	for {
+		compositeKey, dbValue, err := itr.Next()
+		if err != nil {
+			return err
+		}
+		if compositeKey == nil {
+			break
+		}
+		dataKey := encodeDataKey(compositeKey.Namespace, compositeKey.Key)
+		batchSize += len(dataKey) + len(dbValue)
+		dbBatch.Put(dataKey, dbValue)
+		if batchSize >= maxDataImportBatchSize {
+			if err := vdb.db.WriteBatch(dbBatch, true); err != nil {
+				return err
+			}
+		}
+	}
+	return vdb.db.WriteBatch(dbBatch, true)
+}
+
+// IsEmpty return true if the statedb does not have any content
+func (vdb *versionedDB) IsEmpty() (bool, error) {
+	return vdb.db.IsEmpty()
```

Comment:
1. Will the `privacyenabledstate` pkg call this before calling the `ImportState()` so that it can decide on whether to call `Remove()`? 
2. For `statecouchdb`, does it translate to whether metadata DB exist? -- assuming that metadata DB is the first DB to be created during the `ImportState()`?

Reply:
yes, to both (except that for the couch case, the requirement of existence of metadatadb may be relaxed, so long it does not contain any `namespace-dbname` mapping data yet)